colibri/src/icons.rs

```diff
@@ -261,68 +384,73 @@ pub async fn query_icon_remotely(
         _ => None
     } {
         if let Some(icon_bytes) = query_image_from_cdn(url).await {
-            let _ = tokio::fs::write(path.with_extension(extension), &icon_bytes)
-                .await
-                .map_err(|e| {
-                    error!("Unable to write {} to the file system due to {}", path.display(), e);
-                });
-            return;
+            return write_icon_to_file(&path, extension, &icon_bytes).await;
         }
     };
 
-    let (icon_bytes, extension) = match extract_chain_id_and_address(&asset_id) {
-        Some((chain_id, address)) => {
-            match query_token_icon_and_extension(chain_id, &address, SMOLDAPP_BASE_URL).await {
-                Some((bytes, ext)) => (bytes, ext),
-                None => match coingecko.query_asset_image(&asset_id).await {
-                    Some(bytes) => (bytes, "png"),
-                    None => {
-                        debug!(
-                            "Icon not found for asset {}. Writing zero bytes file",
-                            asset_id
-                        );
-                        write_zero_bytes_file(path.as_path()).await;
-                        return;
-                    }
-                },
+    if let Some((chain_id, contract_address, token_id)) = parse_asset_identifier(&asset_id) {
+        let blockchain = match SupportedBlockchain::from_chain_id(chain_id) {
+            Some(blockchain) => blockchain,
+            None => {
```

Comment:
Unless I'm mistaken here you are checking always for on chain nft info but you need only to do this in the case of NFTs at most